server/app/services/UserService.ts

```diff
@@ -64,7 +70,53 @@ export class UserService {
     if (!updatedUser) {
       throw new NotFoundError(`User with username ${username} was not found`);
     }
-
     return UserEntity.buildUser(updatedUser);
   }
+
+  public async forgotPassword(url: string, body: IUser): Promise<string> {
+    const email = body.email;
+    try {
+      const user: IUserEntity = UserEntity.buildUser(await this.userRepo.findByEmail(body));
+      const accessToken = generateToken(user);
+      requestMap.set(email, accessToken);
+
+      const info: SentMessageInfo = await this.systemService.sendEmail({
+        to: [email],
+        subject: 'ERP Password Reset',
+        emailBody: `This is the ERP Team07,\n\nPlease reset your password using the following link: ${url}/reset/${accessToken}\n\nRegards.`,
+      });
+      return info;
+    } catch (err) {
+      throw new NotFoundError(`User with email ${body.email} was not found`);
+    }
+  }
+
+  public async resetPassword(token: string, pass: string): Promise<string> {
+    let email;
+    let found = false;
+
+    for (const [_email, _token] of requestMap.entries()) {
+      if (_token === token) {
+        found = true;
+        email = _email;
+        break;
+      }
+    }
```

Comment:
Replace this for loop with a .get() on the map data structure.

Insert token as key if that's what you want to search for.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map/get